images/origin/scripts/recycler.sh

```diff
@@ -8,36 +8,25 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-function recycle(){
-
-    # first and only arg is the directory to scrub
-    dir=$1
-
-    if [ -z "$dir" ]; then
-        echo "Usage:  scrub_directory some/path/to/scrub"
-        return 1
-    fi
-
-    if [ ! -e $dir ]; then
-        echo "scrub directory $dir does not exist"
-        return 1
-    fi
-
-    # remove everything but keep the directory for re-use as a volume
-    rm -rfv $dir/*
-
-    if [ -z "$(ls -A $dir)" ]; then
-        echo "scrub directory $dir is empty"
-        return 0
-    else
-        echo "scrub directory $dir is not empty"
-        return 1
-    fi
-}
-
-if recycle $1; then
+
+# first and only arg is the directory to scrub
+dir=${1:-}
+
+if [ -z $dir ]; then
+    echo "Usage:  recycler.sh some/path/to/scrub"
+    exit 1
+fi
+
+echo "Scrubbing $dir"
+
+# capture the output, which can contain permission denied errors
+rm_output=$(rm -rf $dir/* 2>&1 || true)
+
+# silence is golden
+if [ -z "$rm_output" ]; then
```

Comment:
Isn't that how I originally had it?   Use errexit and capture the output and test its output for emptiness?